### PR TITLE
Migrate from boost program options to CLI11

### PR DESCRIPTION
**Pull request by jgarciamesa**
- Remove `boost::program_options` dependency.
- Split arguments into options and flags, according to CLI11 syntax.
- Fix `clang-tidy` errors and warnings and format according to `clang-format` syntax.
- Add sublime text configuration file.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -48,4 +48,7 @@ src/config.h
 
 # Ignore the build directory
 build/*
-!build/.gitkeep
+builddir/*
+
```
Comment (reedacartwright) [inline .gitignore:52]: Why did you remove .gitkeep and add builddir to this?

File: contrib/CLI/CLI11.hpp [hunk 2]
```diff
@@ -4843,8 +4843,8 @@ class App {
         set_help_flag("-h,--help", "Print this help message and exit");
     }
 
-    App(const App &) = delete;
-    App &operator=(const App &) = delete;
+    App(const App &) = default;
+    App &operator=(const App &) = default;
```
Comment (reedacartwright) [inline contrib/CLI/CLI11.hpp]: What is the reason for this change?

File: src/dawg.cpp [hunk 3]
```diff
@@ -17,7 +17,8 @@
 */
 #include "dawg.h"
 
-#include <boost/config.hpp>
+// #include <boost/config.hpp>
```
Comment (reedacartwright) [inline src/dawg.cpp]: Is there a reason to keep this commented out code around?

File: src/dawg.cpp [hunk 4]
```diff
@@ -29,9 +30,9 @@
 #include "dawg/trick.h"
 #include "dawg_app.h"
 
-using namespace std;
-using namespace boost;
-using namespace dawg;
+// using namespace std;
+// using namespace boost;
+// using namespace dawg;
```
Comment (reedacartwright) [inline src/dawg.cpp]: Same question as above.

File: .gitignore (hunk 1 above)
Comment (jgarciamesa) [reply .gitignore:52]: I removed the .gitkeep because Meson doesn't require the build folder to exist beforehand, but there's no other advantage besides having one less folder.
I added builddir since it's the default on the Meson documentation and the one I have been using. I can remove it.

File: contrib/CLI/CLI11.hpp (hunk 2 above)
Comment (jgarciamesa) [reply contrib/CLI/CLI11.hpp]: Try to fix a clang tidy warning (clang-diagnostic-defaulted-function-deleted).

`dawg/src/dawg_app.h:32:5: warning: explicitly defaulted copy constructor is implicitly deleted [clang-diagnostic-defaulted-function-deleted]`
`    dawg_app(const dawg_app&) = default;        // copy constructor`

`dawg/src/dawg_app.h:34:5: warning: explicitly defaulted move constructor is implicitly deleted [clang-diagnostic-defaulted-function-deleted]`
`    dawg_app(dawg_app&&) = default;             // move constructor`

File: src/dawg.cpp (hunk 3 above)
Comment (jgarciamesa) [reply src/dawg.cpp]: Not at all, missed it when cleaning up. Removed it on last commit already.

File: src/dawg.cpp (hunk 4 above)
Comment (jgarciamesa) [reply src/dawg.cpp]: Missed it cleaning up, fixed on last commit.

File: .gitignore (hunk 1 above)
Comment (reedacartwright) [reply .gitignore:52]: Can you send me a link to the Meson documentation?

File: contrib/CLI/CLI11.hpp (hunk 2 above)
Comment (reedacartwright) [reply contrib/CLI/CLI11.hpp]: There are good reasons for a library to delete their copy constructors. This warning isn't due toe CLI11. It's due to how you have written `class dawg_app`. Remove those two constructors from dawg_app and revert the changes to CLI11. Let's see if that works.

File: .gitignore (hunk 1 above)
Comment (jgarciamesa) [reply .gitignore:52]: [Mesonbuild documentation - configure build directory](https://mesonbuild.com/Running-Meson.html#configuring-the-build-directory)

Comment (reedacartwright) [reply .gitignore:52]: I think we should make it ignore both `build` and `builddir`.

Comment (eli-schwartz) [reply .gitignore:52]: Simply do not ignore anything.

Since Meson 0.57.0, it creates `build/.gitignore` or `builddir/.gitignore` but also `builddir-release/.gitignore`, `builddir-debug/.gitignore`, `_build/.gitignore`, or anything else you might need.

It creates it with the contents `*` so as to ignore all generated files, you don't need to do anything yourself to set this up.

File: contrib/CLI/CLI11.hpp (hunk 2 above)
Comment (jgarciamesa) [reply contrib/CLI/CLI11.hpp]: I reverted the changes on CLI11.hpp and fixed it on `class dawg_app`. See [last commit](https://github.com/CartwrightLab/dawg/pull/75/commits/dd9d20ee8b99128e12c82b78e6cba7bc6282f0bc).
